test/org/sagebionetworks/bridge/services/StudyServiceMockTest.java

```diff
@@ -1459,6 +1461,47 @@ public void testAllFourTemplatesAreSanitized() {
         assertHtmlTemplateSanitized( study.getVerifyEmailTemplate() );
         assertHtmlTemplateSanitized( study.getAccountExistsTemplate() );
     }
+    
+    @Test
+    public void updateStudyCorrectlyDetectsEmailChangesInvolvingNulls() {
+        // consent email still correctly detected
+        String originalEmail = TestUtils.getValidStudy(StudyServiceMockTest.class).getConsentNotificationEmail();
+        String newEmail = "changed@changed.com";
+        
+        setupConsentEmailChangeTest(null, null);
+        setupConsentEmailChangeTest(originalEmail, originalEmail);
+        setupConsentEmailChangeTest(null, newEmail);
+        setupConsentEmailChangeTest(originalEmail, null);
+        setupConsentEmailChangeTest(originalEmail, newEmail);
+    }
+    
+    private void setupConsentEmailChangeTest(String originalEmail, String newEmail) {
+        reset(sendMailService);
+        Study original = TestUtils.getValidStudy(StudyServiceMockTest.class);
+        original.setConsentNotificationEmail(originalEmail);
+        when(studyDao.getStudy(any())).thenReturn(original);
+        
+        Study update = TestUtils.getValidStudy(StudyServiceMockTest.class);
+        update.setConsentNotificationEmail(newEmail);
+        // just assume this is true for the test so defaults aren't set
+        update.setConsentNotificationEmailVerified(true);
+        
+        service.updateStudy(update, true);
+        
+        boolean hasChanged = !Objects.equals(original.getConsentNotificationEmail(),
+                update.getConsentNotificationEmail());
+        
+        if (update.getConsentNotificationEmail() == null) {
+            verify(sendMailService, never()).sendEmail(any());
+            assertEquals(hasChanged, !update.isConsentNotificationEmailVerified());
```

Comment:
update.isConsentNotificationEmailVerified() is always the opposite of hasChanged. Would it be better to pull this out of the if-else block?

Reply:
Created two separate blocks